packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L1StandardBridge.sol

```diff
@@ -0,0 +1,318 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: ovm
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Interface Imports */
+import { iOVM_L1StandardBridge } from "../../../iOVM/bridge/tokens/iOVM_L1StandardBridge.sol";
+import { iOVM_L1ERC20Bridge } from "../../../iOVM/bridge/tokens/iOVM_L1ERC20Bridge.sol";
+import { iOVM_L2ERC20Bridge } from "../../../iOVM/bridge/tokens/iOVM_L2ERC20Bridge.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/* Library Imports */
+import { OVM_CrossDomainEnabled } from "../../../libraries/bridge/OVM_CrossDomainEnabled.sol";
+import { Lib_PredeployAddresses } from "../../../libraries/constants/Lib_PredeployAddresses.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+/**
+ * @title OVM_L1StandardBridge
+ * @dev The L1 ETH and ERC20 Bridge is a contract which stores deposited L1 funds and standard tokens that are in use on L2.
+ * It synchronizes a corresponding L2 Bridge, informing it of deposits, and listening to it for newly finalized withdrawals.
+ *
+ * Compiler used: solc
+ * Runtime target: EVM
+ */
+contract OVM_L1StandardBridge is iOVM_L1StandardBridge, OVM_CrossDomainEnabled {
+    using SafeMath for uint;
+    using SafeERC20 for IERC20;
+
+    /********************************
+     * External Contract References *
+     ********************************/
+
+    address public l2TokenBridge;
+
+    // Maps L1 token to L2 token to balance of the L1 token deposited
+    mapping(address => mapping (address => uint256)) public deposits;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    // This contract lives behind a proxy, so the constructor parameters will go unused.
+    constructor()
+        OVM_CrossDomainEnabled(address(0))
+    {}
+
+    /******************
+     * Initialization *
+     ******************/
+
+    /**
+     * @param _l1messenger L1 Messenger address being used for cross-chain communications.
+     * @param _l2TokenBridge L2 standard bridge address.
+     */
+    function initialize(
+        address _l1messenger,
+        address _l2TokenBridge
+    )
+        public
+    {
+        require(messenger == address(0), "Contract has already been initialized.");
+        messenger = _l1messenger;
+        l2TokenBridge = _l2TokenBridge;
+    }
+
+    /**************
+     * Depositing *
+     **************/
+
+    /// @dev Modifier requiring sender to be EOA.  This check could be bypassed by a malicious contract via initcode, but it takes care of the user error we want to avoid.
+    modifier onlyEOA() {
+        // Used to stop deposits from contracts (avoid accidentally lost tokens)
+        require(!Address.isContract(msg.sender), "Account not EOA");
```

Comment:
TODO: Make a ticket for the general problem that makes this `require` statement necessary